tests/unit_tests/db_engine_specs/test_mssql.py

```diff
@@ -454,3 +454,25 @@ def test_denormalize_name(name: str, expected_result: str):
     from superset.db_engine_specs.mssql import MssqlEngineSpec as spec
 
     assert spec.denormalize_name(mssql.dialect(), name) == expected_result
+
+
+@pytest.mark.parametrize(
+    "native_type,sqla_type,generic_type",
+    [
+        ("uniqueidentifier", UUID, GenericDataType.STRING),  # Test case for UUID
```

Comment:
There's already a parametrized test for this. Please move this to that test to avoid repeating the test boilerplate.

Reply:
I actually had sqlalchemy 2.x in mind while I was writing this, so instead created a TypeDecorator.

Also refactored the test into the existing function, apologies for missing it.